scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Test for an order of dependencies in setup.py
+"""
+import os
+import sys
+from typing import List
+
+from rich import print
+
+errors = []
+
+MY_DIR_PATH = os.path.dirname(__file__)
+SOURCE_DIR_PATH = os.path.abspath(os.path.join(MY_DIR_PATH, os.pardir, os.pardir, os.pardir))
+
+
+def _check_list_sorted(the_list: List[str], message: str) -> None:
```

Comment:
Do you consider to use [`difflib`](https://docs.python.org/3/library/difflib.html)?

Reply:
Prefer to keep in sync with other checks. But maybe rewriting them all to difflib is a good idea to do @mik-laj  :)

Reply:
Feel free as a follow-up (for all pre-commits)

Reply:
Ok. Will do.